File: backend/src/main/java/com/devmatch/backend/domain/project/mapper/ProjectMapper.java
package com.devmatch.backend.domain.project.mapper;

import com.devmatch.backend.domain.project.dto.ProjectDetailResponse;
import com.devmatch.backend.domain.project.entity.Project;
import java.util.Arrays;

public class ProjectMapper {

  public static ProjectDetailResponse toProjectDetailResponse(Project project) {
    return new ProjectDetailResponse(
        project.getId(),
        project.getTitle(),
        project.getDescription(),
        Arrays.stream(project.getTechStack().split(", ")).toList(),
        project.getTeamSize(),
        project.getCurrentTeamSize(),
        project.getCreator().getNickName(),
        project.getStatus().name(),
        project.getContent(),
        project.getDurationWeeks(),
        project.getCreatedAt()
    );
  }
}

File: backend/src/main/java/com/devmatch/backend/domain/project/mapper/ProjectMapper.kt
package com.devmatch.backend.domain.project.mapper

import com.devmatch.backend.domain.project.dto.ProjectDetailResponse
import com.devmatch.backend.domain.project.entity.Project

object ProjectMapper {
    
    fun toProjectDetailResponse(project: Project): ProjectDetailResponse {
        return ProjectDetailResponse(
            project.id,
            project.title,
            project.description,
            project.techStack.split(", "),
            project.teamSize,
            project.currentTeamSize,
            project.creator.nickname,
            project.status.name,
            project.content,
            project.durationWeeks,
            project.createdAt
        )
    }
}
